client/src/partials/vouchers/toolkit.service.js

```diff
@@ -0,0 +1,49 @@
+angular.module('bhima.services')
+  .service('VoucherToolkitService', VoucherToolkitService);
+
+VoucherToolkitService.$inject = ['$http', '$uibModal', 'util'];
+
+function VoucherToolkitService($http, Modal, util) {
+  var service = this;
+
+  // expose the service
+  service.open = open;
+
+  // tools of advanced journal voucher
+  service.tools = {
+    // this tool help to import transaction rows for a convention payment
+    'convention_payment' : {
+      label : 'VOUCHERS.GLOBAL.CONVENTION_INVOICES',
+      controller : 'ConventionPaymentKitController',
+      templateUrl : 'partials/vouchers/toolkit/convention_payment.modal.html'
+    }
+  };
+
+  // service options
+  service.options = Object.keys(service.tools).map(function (item) {
+    var tool = service.tools[item];
+    return tool;
+  });
+
+  /**
+   * @function open
+   * @description This function initialise a modal for the selected tool
+   * @param {Object} option A detailed option object that should provide
+   * label, controller string, templateUrl string
+   */
+  function open(option) {
+    if (!option.controller || !option.templateUrl) { return; }
+
+    var controllerString = option.controller.concat(' as ToolCtrl');
```

Comment:
Instead of doing this...

You can just use the property `controllerAs` in the options.  [Here is the documentation](https://angular-ui.github.io/bootstrap/#/-uibmodal-s-open-function).

```js
var instance = Modal.open({
  controller : 'SomeController',
  controllerAs: '$ctrl'
});
```